py-polars/polars/io/pyarrow_dataset/anonymous_scan.py

```diff
@@ -96,6 +109,12 @@ def _scan_pyarrow_dataset_impl(
     common_params = {"columns": with_columns, "filter": _filter}
     if batch_size is not None:
         common_params["batch_size"] = batch_size
+    if pyarrow_options:
+        for pyarrow_option_name, pyarrow_option in pyarrow_options.items():
+            if pyarrow_option_name in common_params:
+                error = f"Tried to overwrite already set pyarrow parameter {pyarrow_option_name}"
+                raise ValueError(error)
+            common_params[pyarrow_option_name] = pyarrow_option
```

Comment:
You don't need to write the logic for this. If you just pass `**pyarrow_options` to the `to_table` call, you will get a TypeError if the same option was passed twice.